controllers/artefacts.rb

```diff
@@ -86,48 +96,52 @@ class ArtefactsController < ApplicationController
                 end
                 resouces_count += props_count
 
-                reply Goo::Base::Page.new(page, size, resouces_count, resources)
+                reply page_object(resources, resouces_count)
             end
-          
+
             get '/classes' do
                 ontology, latest_submission = get_ontology_and_latest_submission
+                check_access(ontology)
                 type = LinkedData::Models::Class.class_rdf_type(latest_submission)
                 attributes, page, size = settings_params(LinkedData::Models::Class).first(3)
                 
                 if type == RDF::OWL[:Class]
                     reply handle_resources_request(ontology, latest_submission, LinkedData::Models::Class, attributes, page, size)
                 else
-                    reply empty_page(page, size)
+                    reply empty_page
                 end
             end
           
             get '/concepts' do
                 ontology, latest_submission = get_ontology_and_latest_submission
+                check_access(ontology)
                 type = LinkedData::Models::Class.class_rdf_type(latest_submission)
                 attributes, page, size = settings_params(LinkedData::Models::Class).first(3)
                 
                 if type.to_s == "http://www.w3.org/2004/02/skos/core#Concept"
                     reply handle_resources_request(ontology, latest_submission, LinkedData::Models::Class, attributes, page, size)
                 else
-                    reply empty_page(page, size)
+                    reply empty_page
                 end
             end
           
             get '/properties' do
                 ontology, latest_submission = get_ontology_and_latest_submission
+                check_access(ontology)
                 _, page, size = settings_params(LinkedData::Models::OntologyProperty).first(3)
                 props_page, _ = handle_properties_request(ontology, latest_submission, page, size)
                 reply props_page
             end
 
             get '/individuals' do
                 ontology, latest_submission = get_ontology_and_latest_submission
+                check_access(ontology)
                 type = LinkedData::Models::Class.class_rdf_type(latest_submission)
                 attributes, page, size = settings_params(LinkedData::Models::Instance).first(3)
                 if type == RDF::OWL[:Class]
                     reply handle_resources_request(ontology, latest_submission, LinkedData::Models::Instance, attributes, page, size)
                 else
-                    reply empty_page(page, size)
+                    reply empty_page
                 end
             end
```

Comment:
in line 148, why you do add individuals,properties,concepts, classes ?

Reply:
yes we can regroup them like this:

- `[concepts schemes collections labels]` => when type is skos concept
- `[classes individuals]` => when type is owl